app/models/experimentation/notifications_experiment.rb

```diff
@@ -95,10 +95,20 @@ def evict_patients
 
     def mark_visits
       treatment_group_memberships.status_enrolled
-        .select("distinct on (treatment_group_memberships.patient_id) treatment_group_memberships.*, bp.id bp_id, bs.id bs_id, pd.id pd_id")
-        .joins("left outer join blood_pressures bp on bp.patient_id = treatment_group_memberships.patient_id AND bp.recorded_at > experiment_inclusion_date")
-        .joins("left outer join blood_sugars bs on bs.patient_id = treatment_group_memberships.patient_id AND bs.recorded_at > experiment_inclusion_date")
-        .joins("left outer join prescription_drugs pd on pd.patient_id = treatment_group_memberships.patient_id AND pd.device_created_at > experiment_inclusion_date")
+        .select("distinct on (treatment_group_memberships.patient_id) treatment_group_memberships.*,
+                 bp.id bp_id, bs.id bs_id, pd.id pd_id")
+        .joins("left outer join blood_pressures bp
+          on bp.patient_id = treatment_group_memberships.patient_id
+          and bp.recorded_at > experiment_inclusion_date
+          and bp.deleted_at is null")
+        .joins("left outer join blood_sugars bs
+          on bs.patient_id = treatment_group_memberships.patient_id
+          and bs.recorded_at > experiment_inclusion_date
+          and bs.deleted_at is null")
+        .joins("left outer join prescription_drugs pd
+          on pd.patient_id = treatment_group_memberships.patient_id
+          and pd.device_created_at > experiment_inclusion_date
+          and pd.deleted_at is null")
```

Comment:
Can we do this in a more railsy way? Wouldn't `has_many :blood_sugars, primary_key: :patient_id` get the discarded ones out by default?

Reply:
Also, this needs a test.

Reply:
`has_many` doesn't help with the join conditions. There isn't a more railsy way to do this afaict :/

Reply:
I tried the above snippet before posting it actually. And postgres (in this case at least) won't differentiate between the condition on a `join` and a `where`. It's not critical though, the sql is fairly readable still.

Reply:
Could do an OR in the where no? 🤷